app/Services/Geolocator/IPGeolocationApi.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+namespace App\Services\Geolocator;
+
+use Illuminate\Support\Facades\Http;
+
+class IPGeolocationApi implements GeoLocatorApiInterface
+{
+    private  $ip;
+    public function __construct($ip)
+    {
+        $this->ip = $ip;
+    }
+    public  function getDetails()
+    {
+        $ipInfo = Http::get("http://ip-api.com/json/{$this->ip}");
```

Comment:
Move this to the services config file and add it to the .env file with this value as the default in the services config file services.php